diff -ruN Mail-SPF-Query-1.997.orig/spfd Mail-SPF-Query-1.997.host/spfd
--- Mail-SPF-Query-1.997.orig/spfd	2004-01-14 04:40:26.000000000 +0100
+++ Mail-SPF-Query-1.997.host/spfd	2005-06-10 14:24:12.000000000 +0200
@@ -17,12 +17,15 @@
 #    spfd [OPTION]...
 #
 # DEFAULT
-#    spfd -port=5970
+#    spfd -port=5970 -host 127.0.0.1
 #
 # OPTIONS
 #    -port=PORTNUM
 #        Listen with a tcp socket on port PORTNUM (mutually exlusive with -path)
 #
+#    -host=LOCALADDR
+#        Listen on LOCALADDR IP (mutually exlusive with -path)
+#
 #    -path=FILESPEC
 #        Listen with a unix socket at FILESPEC (mutually exlusive with -port)
 #
@@ -64,17 +67,22 @@
 use Socket;
 
 use strict;
-use vars qw($port $path $pathuser $pathgroup $pathmode $setuser $setgroup);
+use vars qw($port $host $path $pathuser $pathgroup $pathmode $setuser $setgroup);
 
 # FIXME
 sub usage () {
-  print "usage: spfd ( -port=5970 | -path=/var/spfd ) [-setuser=(uid|username)] [-setgroup=(gid|groupname)]\n";
+  print "usage: spfd ( -port=5970 [-host=ip] | -path=/var/spfd ) [-setuser=(uid|username)] [-setgroup=(gid|groupname)]\n";
   print "usage:      [ -pathuser=(uid|username)] [ -pathgroup=(gid|groupname)] [-pathmode=mode]\n";
-  print "usage: spfd assuming -port=5970\n";
+  print "usage: spfd assuming -port=5970 -host 127.0.0.1\n";
 }
 
 sub DEBUG () { $ENV{DEBUG} }
 
+if (not $host and not $path) {
+  usage;
+  $host='127.0.0.1';
+}
+
 if (not $port and not $path) {
   usage;
   $port=5970;
@@ -93,12 +101,12 @@
 if ($port) {
   $sock_type = "inet";
   @args = (Listen    => 1,
-	   LocalAddr => "127.0.0.1",
+	   LocalAddr => $host,
 	   LocalPort => $port,
 	   ReuseAddr => 1
 	   );
-  print "$$ will listen on $port\n";
-  $0 = "spfd listening on $port";
+  print "$$ will listen on $host:$port\n";
+  $0 = "spfd listening on $host:$port";
 } elsif ($path) {
   $sock_type = "unix";
   unlink $path if -S $path;
